src/structs/lepton_encoder.rs

```diff
@@ -654,58 +695,70 @@ fn roundtrip_ones() {
 #[test]
 fn roundtrip_large_coef() {
-    let left = AlignedBlock::new([1023; 64]);
-    let above = AlignedBlock::new([1023; 64]);
-    let here = AlignedBlock::new([1023; 64]);
-
-    // verified output from a previous run. If this fails, then you've made a change
-    // to the binary format of the compressor and probably shouldn't do that since then
-    // the compressor and decompressor won't be compatible anymore.
-    let verified_output = [
-        0x506f55523369cf48,
-        0x5b2795bc24a04d2e,
-        0xdcb68ed904cfc4f9,
-        0x80efab28c62db071,
-    ];
-
-    roundtrip_read_write_coefficients_all(
-        &left,
-        &above,
-        &here,
-        &verified_output,
+    let block = AlignedBlock::new([1023; 64]);
+
+    roundtrip_read_write_coefficients(
+        &block,
+        &block,
+        &block,
+        &block,
+        [1; 64],
+        0xcf268c76674a5c83,
+        &EnabledFeatures::compat_lepton_scalar_read(),
+    );
+
+    // now test with maximum quantization table. In theory this is legal according
+    // the JPEG format and there is no code preventing this from being attempted
+    // by the encoder.
+
+    roundtrip_read_write_coefficients(
+        &block,
+        &block,
+        &block,
+        &block,
+        [65535; 64],
+        0xc4410bb82397c41f,
         &EnabledFeatures::compat_lepton_scalar_read(),
     );
 }
 
+/// "random" set of blocks to ensure that all ranges of coefficients work properly
 #[test]
-fn roundtrip_random() {
-    use rand::rngs::StdRng;
+fn roundtrip_random_seed() {
     use rand::Rng;
-    use rand::SeedableRng;
 
-    let mut rng = StdRng::from_seed([2u8; 32]);
+    // the 76 seed is a choice that doesn't overflow the DC coefficient
+    // since the encoder is somewhat picky if the DC estimate overflows
+    // it also has different behavior for 32 and 16 bit codepath
+    let mut rng = crate::helpers::get_rand_from_seed([76; 32]);
 
     let arr = [0i16; 64];
 
-    let left = AlignedBlock::new(arr.map(|_| rng.gen_range(-1023..=1023)));
-    let above = AlignedBlock::new(arr.map(|_| rng.gen_range(-1023..=1023)));
-    let here = AlignedBlock::new(arr.map(|_| rng.gen_range(-1023..=1023)));
-
-    // verified output from a previous run. If this fails, then you've made a change
-    // to the binary format of the compressor and probably shouldn't do that since then
-    // the compressor and decompressor won't be compatible anymore.
-    let verified_output = [
-        0x4b08a910feb758e8,
-        0x44c3f76e93f1d204,
-        0x899bc6e64957e400,
-        0x322e78e37fd7ed13,
-    ];
-
-    roundtrip_read_write_coefficients_all(
+    let left = AlignedBlock::new(arr.map(|_| rng.gen_range(-2047..=2047)));
+    let above = AlignedBlock::new(arr.map(|_| rng.gen_range(-2047..=2047)));
+    let here = AlignedBlock::new(arr.map(|_| rng.gen_range(-2047..=2047)));
+    let above_left = AlignedBlock::new(arr.map(|_| rng.gen_range(-2047..=2047)));
```

Comment:
```suggestion
    let mut left = AlignedBlock::new(arr.map(|_| rng.gen_range(-2047..=2047)));
    let mut above = AlignedBlock::new(arr.map(|_| rng.gen_range(-2047..=2047)));
    let mut here = AlignedBlock::new(arr.map(|_| rng.gen_range(-2047..=2047)));
    let mut above_left = AlignedBlock::new(arr.map(|_| rng.gen_range(-2047..=2047)));
    left.set_coefficient(0, rng.gen_range(-1024..=1023));
    above.set_coefficient(0, rng.gen_range(-1024..=1023));
    here.set_coefficient(0, rng.gen_range(-1024..=1023));
    above_left.set_coefficient(0, rng.gen_range(-1024..=1023));
```

As I've shown in [the example here](https://github.com/Melirius/lepton_jpeg_rust/commit/7d80986cd7d40b4118b1916823eb5de3c7a360c7), `adv_predict_or_unpredict_dc` breaks outside of this range and encoder will give an error on such a file, so values outside of it cannot appear in Lepton.

Reply:
it actually breaks in multiple cases, both when the coefficient is out of the range, but also if the predictor is out of the range. Kind of annoying. I just ended up choosing a random seed that generated things that were within range.  

I think the original idea was that what was sent to the encode was the delta between the predictor and actual DC, but there's a bug in the way that it adds the adjustment so effectively it rejects both if the DC is out of range, but also if the delta is out of range.